### PR TITLE
Memory leak fixes

### PR DESCRIPTION
**Pull request by robertguetzkow**
This pull request fixes several memory leaks.

#### Check all that apply (change to `[x]`)
- [ ] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [ ] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [x] This is a minor change.

### CODE REVIEW

**Comment by robertguetzkow** [comment]
The improvements should be done now. Made some painfully obvious mistakes in the first commits because having a cold & coding don't match, sorry for that.

**Comment by robertguetzkow** [comment]
AppVeyor fails because it doesn't finish within 60 minutes and aborts the build process.

**Comment by jdumas** [comment]
Yeah it's annoying... It's a hard limit of AppVeyor and I don't know how we can speed up compilation of the Windows build. I'll restart it until it goes through. Thanks for your patience!

File: include/igl/principal_curvature.cpp [hunk 1]
```diff
@@ -522,8 +522,12 @@ IGL_INLINE void CurvatureCalculator::getSphere(const int start, const double r,
       }
     }
   }
-  free(extra_candidates);
-  free(queue);
+  while(!extra_candidates->empty()){
```
Comment (jdumas) [inline include/igl/principal_curvature.cpp]: I believe this while and the clear are unnecessary. The `delete` operator should call the destructor of both variables. Honestly looking at the code you might as well remove the `new` operators altogether (and this is also true for the `bool* visited` which is `calloc`ed.

Comment (robertguetzkow) [reply include/igl/principal_curvature.cpp]: [You're right](https://en.cppreference.com/w/cpp/container/priority_queue/~priority_queue), my C++ skills are a bit rusty. I'll commit the changes to the function. Unnecessary allocations with `new` happen in other code as well, which may need to be fixed in the future.

Comment (robertguetzkow) [reply include/igl/principal_curvature.cpp]: BTW a potential fix for the slow AppVeyor build could be `/m:2` in the `MSBuildOptions` to use two threads for the two cores available in the VMs.

**Comment by robertguetzkow** [comment]
I'll revert the change for the bool array, because it requires a size known at runtime which isn't support by C++ standard (as far as I'm aware). So it has to be allocated or we change it to a vector.

File: include/igl/principal_curvature.cpp [hunk 2]
```diff
@@ -448,7 +448,8 @@ IGL_INLINE void CurvatureCalculator::getKRing(const int start, const double r, s
   int bufsize=vertices.rows();
   vv.reserve(bufsize);
   std::list<std::pair<int,int> > queue;
-  bool* visited = (bool*)calloc(bufsize,sizeof(bool));
+  std::vector<bool> visited;
```
Comment (jdumas) [inline include/igl/principal_curvature.cpp]: Can you initialize it to false as well? Using the constructor `std::vector<bool> visited(bufsize, false);`

**Comment by robertguetzkow** [comment]
Are there any additional changes that you need before merging?

**Pull request merged**